### PR TITLE
soc/integration: ensure CSR constants are in uppercase

### PR DESCRIPTION
**Pull request by gsomlo**
Fixup over commit 8be5824e.

Signed-off-by: Gabriel Somlo <gsomlo@gmail.com>

### CODE REVIEW

File: litex/soc/integration/soc_core.py [hunk 1]
```diff
@@ -445,7 +445,8 @@ def do_finalize(self):
 
         # Add CSRs / Config items to constants
         for name, constant in self.csrbankarray.constants:
-            self.add_constant(name + "_" + constant.name, constant.value.value)
+            self.add_constant(name.upper() + "_" + constant.name.upper(),
+                              constant.value.value)
```
Comment (mithro) [inline litex/soc/integration/soc_core.py]: Should `constant.value.value.upper()`?

**Comment by gsomlo** [comment]
On Tue, Oct 01, 2019 at 09:37:45AM -0700, Tim Ansell wrote:
> @mithro commented on this pull request.
> 
> ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
> 
> In litex/soc/integration/soc_core.py:
> 
> > @@ -445,7 +445,8 @@ def do_finalize(self):
> 
>          # Add CSRs / Config items to constants
>          for name, constant in self.csrbankarray.constants:
> -            self.add_constant(name + "_" + constant.name, constant.value.value)
> +            self.add_constant(name.upper() + "_" + constant.name.upper(),
> +                              constant.value.value)
> 
> Should constant.value.value.upper()?

Hmmm... I'd expect not, since it's supposed to be numeric, no?

**Comment by daveshah1** [comment]
There might also be some cases of string literals in constants, which you would probably want to preserve case rather than convert to uppercase

**Comment by enjoy-digital** [comment]
Thanks, let's preserve case for the value.

**Pull request merged**